### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "options.agent" property must be one of Agent-like Object, undefined, or false. Received an instance of Object

### ISSUE THREAD

**Issue by System-Glitch** [jdalrymple/gitbeaker#1161]
**Description**

I am building a MS Teams bot making a bridge between Teams and our self-hosted Gitlab. When I try to use the API while inside a dialog, I get this error:
```
TypeError [ERR_INVALID_ARG_TYPE]: The "options.agent" property must be one of Agent-like Object, undefined, or false. Received an instance of Object
```

**Steps to reproduce**

```ts
import { Dialog, DialogContext, DialogTurnResult } from "botbuilder-dialogs";
import { Pipelines } from '@gitbeaker/node';
import { IPipelineAction } from "../IPipelineAction";

export default class RunPipelineDialog extends Dialog {
    constructor(dialogId: string) {
        super(dialogId);
    }

    public async beginDialog(context: DialogContext, options?: any): Promise<DialogTurnResult> {
        const data: IPipelineAction = context.context.activity.value;
        console.log(data);

        const api = new Pipelines({
            host: process.env.GITLAB_HOST, // host starts with "https://"
            token: process.env.GITLAB_TOKEN
        });
        await api.showJobs(data.project, +data.pipeline).then((jobs) => {
            console.log(jobs);
        }).catch(console.error);
        return await context.endDialog();
    }
}
```
The project was generated using the [yeoman teams bot generator](https://github.com/pnp/generator-teams).

This is weird because the following piece of code works perfectly when executed alone, not in a bot context. (`node thefile.ts`)
```ts
const Gitlab = require('@gitbeaker/node');

const pipelinesAPI = new Gitlab.Pipelines({
    host: "https://selfhostedgitlabhost",
    token: "my_access_token"
});

pipelinesAPI.showJobs("projectname", 25112).then((jobs) => {
    console.log(jobs);
}).catch(console.error);
```

**Expected behaviour**

The API should also work in a dialog context.

**Actual behaviour**

```
TypeError [ERR_INVALID_ARG_TYPE]: The "options.agent" property must be one of Agent-like Object, undefined, or false. Received an instance of Object
    at new ClientRequest (_http_client.js:136:11)
    at Object.request (http.js:50:10)
    at get (./node_modules/got/index.js:33:18)
    at requestAsEventEmitter (./node_modules/got/index.js:85:2)
    at ./node_modules/got/index.js:91:14
    at new Promise (<anonymous>)
    at asPromise (./node_modules/got/index.js:90:9)
    at got (./node_modules/got/index.js:302:10)
    at ./dist/server.js:30:1570
    at ./dist/server.js:30:1377 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

**Possible fixes**

I have no idea how to fix that it looks like the agent is not set correctly [here](https://github.com/jdalrymple/gitbeaker/blob/98955c1c5efb9b49e675a6500637a3a93abd4d97/packages/gitbeaker-node/src/GotRequester.ts#L25).

**Comment by jdalrymple** [comment]
Hmm, looks like the got properties changed.

**Comment by System-Glitch** [comment]
So this is probably an issue with either `got` or the MS Bot framework?

**Comment by jdalrymple** [comment]
Looks like a missed update in got! Creating a canary build now so you can check it out

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`24.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/24.1.0) :rocket:
